types/react-flatpickr/index.d.ts

```diff
@@ -4,13 +4,19 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
 
-import { Component } from 'react';
-import { EventCallback, Options } from 'flatpickr';
+import { Component, DetailedHTMLProps, InputHTMLAttributes, HTMLAttributes } from 'react';
+import { Hook, Options } from 'flatpickr';
 
-export interface DateTimePickerProps {
+type HTMLProps =
+    {onChange?: Hook}       // needed because there is a conflit with div and input props
```

Comment:
I don't think you need this here, but I'll confirm once we sort out the `flatpickr` issue (mentioned above).